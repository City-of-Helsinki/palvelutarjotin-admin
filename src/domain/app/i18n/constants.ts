export enum VALIDATION_MESSAGE_KEYS {
  EMAIL = 'form.validation.string.email',
  NUMBER_MAX = 'form.validation.number.max',
  NUMBER_MIN = 'form.validation.number.min',
  NUMBER_REQUIRED = 'form.validation.number.required',
  STRING_MAX = 'form.validation.string.max',
  STRING_MIN = 'form.validation.string.min',
  STRING_REQUIRED = 'form.validation.string.required',
  URL = 'form.validation.string.url',
}
