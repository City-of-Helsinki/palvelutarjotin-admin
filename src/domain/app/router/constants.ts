import type { BrowserRouterProps, MemoryRouterProps } from 'react-router-dom';

export const DEFAULT_ROUTER_PROPS = {
  future: {
    v7_startTransition: true,
  },
} as const satisfies BrowserRouterProps satisfies MemoryRouterProps;
