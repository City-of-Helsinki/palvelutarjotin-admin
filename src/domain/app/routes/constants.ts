export enum ROUTES {
  CALLBACK = '/callback',
  CREATE_EVENT = '/events/create',
  CREATE_OCCURRENCE = '/events/:id/occurrences/create',
  EDIT_OCCURRENCE = '/events/:id/occurrences/:occurrenceId/edit',
  EDIT_EVENT = '/events/:id/edit',
  ENROLLMENTS = '/events/:id/occurrences/:occurrenceId/enrollments',
  EVENT_DETAILS = '/events/:id',
  HOME = '/',
  OCCURRENCES = '/events/:id/occurrences',
  SILENT_CALLBACK = '/silent-callback',
}
