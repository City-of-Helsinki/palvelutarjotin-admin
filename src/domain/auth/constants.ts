import { LoginProviderProps } from 'hds-react';

import AppConfig from '../app/AppConfig';

// todo: delete this when new login implemented
export const API_TOKEN_ACTIONS = {
  FETCH_TOKEN_ERROR: 'FETCH_TOKEN_ERROR',
  FETCH_TOKEN_SUCCESS: 'FETCH_TOKEN_SUCCESS',
  RESET_API_TOKEN_DATA: 'RESET_API_TOKEN_DATA',
  START_FETCHING_TOKEN: 'START_FETCHING_TOKEN',
  TOKEN_FETCHED: 'TOKEN_FETCHED',
};

const providerPropertiesBase: LoginProviderProps = {
  userManagerSettings: {
    authority: AppConfig.oidcAuthority,
    client_id: AppConfig.oidcClientId,
    scope: AppConfig.oidcScope,
    response_type: AppConfig.oidcReturnType,
    redirect_uri: `${AppConfig.origin}/callback`,
    automaticSilentRenew: AppConfig.oidcAutomaticSilentRenew,
    silent_redirect_uri: `${AppConfig.origin}/silent_renew.html`,
    post_logout_redirect_uri: `${AppConfig.origin}/`,
  },
  debug: false,
  sessionPollerSettings: {
    pollIntervalInMs: AppConfig.oidcSessionPollerIntervalInMs,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Tunnistamo.
 *
 * @deprecated LoginProviderProps for Tunnistamo integration are deprecated,
 * because the Tunnistamo instance will be changed to
 * the Helsinki-Profile Keycloak instance in the near future.
 */
export const tunnistamoPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}api-tokens/`,
    maxRetries: 10,
    retryInterval: 1000,
  },
};

/**
 * HDS LoginProvider configuration when the Kultus Admin UI
 * is integrated with Keycloak.
 */
export const keycloakPoviderProperties: LoginProviderProps = {
  ...providerPropertiesBase,
  apiTokensClientSettings: {
    url: `${AppConfig.oidcAuthority}protocol/openid-connect/token`,
    maxRetries: 10,
    retryInterval: 1000,
    queryProps: {
      grantType: 'urn:ietf:params:oauth:grant-type:uma-ticket',
      permission: '#access',
    },
    audiences: AppConfig.oidcAudiences,
  },
};
