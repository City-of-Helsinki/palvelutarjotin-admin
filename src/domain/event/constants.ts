export enum EVENT_LANGUAGES {
  AR = 'ar',
  EN = 'en',
  FI = 'fi',
  RU = 'ru',
  SV = 'sv',
  ZH_HANS = 'zh_hans',
}
