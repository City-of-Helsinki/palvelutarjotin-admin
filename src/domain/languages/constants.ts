import { type Language, LanguageCodeEnum } from 'react-helsinki-headless-cms';

export const CMS_FINNISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.Fi,
  id: 'TGFuZ3VhZ2U6Zmk=',
  locale: 'fi',
  name: 'Suomi',
  slug: 'fi',
  __typename: 'Language',
} as const satisfies Language;

export const CMS_ENGLISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.En,
  id: 'TGFuZ3VhZ2U6ZW4=',
  locale: 'en_US',
  name: 'English',
  slug: 'en',
  __typename: 'Language',
} as const satisfies Language;

export const CMS_SWEDISH_LANGUAGE_ENTRY = {
  code: LanguageCodeEnum.Sv,
  id: 'TGFuZ3VhZ2U6c3Y=',
  locale: 'sv_SE',
  name: 'Svenska',
  slug: 'sv',
  __typename: 'Language',
} as const satisfies Language;

export const HARDCODED_CMS_LANGUAGES = [
  { ...CMS_FINNISH_LANGUAGE_ENTRY },
  { ...CMS_ENGLISH_LANGUAGE_ENTRY },
  { ...CMS_SWEDISH_LANGUAGE_ENTRY },
] as const;

/** Data taken from production CMS languages query response. */
export const HARDCODED_CMS_LANGUAGE_QUERY_RESPONSE = {
  data: {
    languages: [...HARDCODED_CMS_LANGUAGES],
  },
} as const;
