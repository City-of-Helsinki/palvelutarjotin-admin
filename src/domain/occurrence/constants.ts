import { TimeAndLocationFormFields } from './types';

export enum OCCURRENCE_URL_PARAMS {
  ENROLMENT_UPDATED = 'enrolmentUpdated',
}

export const defaultInitialValues: TimeAndLocationFormFields = {
  location: '',
  isVirtual: false,
  enrolmentNeeded: false,
  externalEnrolment: false,
  externalEnrolmentUrl: '',
  enrolmentStart: null,
  enrolmentEndDays: '',
  neededOccurrences: '',
  autoAcceptance: true,
  locationDescription: {
    fi: '',
    en: '',
    sv: '',
  },
  hasClothingStorage: false,
  hasSnackEatingPlace: false,
  outdoorActivity: false,
  hasToiletNearby: false,
  hasAreaForGroupWork: false,
  hasIndoorPlayingArea: false,
  hasOutdoorPlayingArea: false,
};
