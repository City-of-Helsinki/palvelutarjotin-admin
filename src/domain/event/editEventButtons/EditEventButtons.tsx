import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory } from 'react-router';

import BackButton from '../../../common/components/backButton/BackButton';
import { SUPPORT_LANGUAGES } from '../../../constants';
import { EventQuery } from '../../../generated/graphql';
import { Language } from '../../../types';
import { ROUTES } from '../../app/routes/constants';
import EventLanguageSelector from '../eventLanguageSelector/EventLanguageSelector';
import styles from './editEventButtons.module.scss';

interface Props {
  dirty: boolean;
  eventData?: EventQuery;
  onClickLanguage: (language: Language) => void;
  selectedLanguage: Language;
}

const EditEventButtons: React.FC<Props> = ({
  dirty,
  eventData,
  onClickLanguage,
  selectedLanguage,
}) => {
  const { t } = useTranslation();
  const history = useHistory();

  const moveToEventList = () => {
    history.push(ROUTES.HOME);
  };

  return (
    <div className={styles.editEventButtons}>
      <div className={styles.backButtonWrapper}>
        <BackButton onClick={moveToEventList}>
          {t('editEvent.buttons.buttonBack')}
        </BackButton>
        {dirty && (
          <span className={styles.dirtyText}>
            {t('editEvent.buttons.textDirty')}
          </span>
        )}
      </div>
      <EventLanguageSelector
        languages={Object.values(SUPPORT_LANGUAGES).map((language) => ({
          isCompleted: !!eventData?.event?.name[language],
          isDisabled: false,
          label: t(`common.languages.${language}`),
          value: language,
        }))}
        onClick={onClickLanguage}
        selectedLanguage={selectedLanguage}
      />
    </div>
  );
};

export default EditEventButtons;
