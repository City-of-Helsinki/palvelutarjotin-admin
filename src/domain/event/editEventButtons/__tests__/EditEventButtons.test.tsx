import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Router, { MemoryRouter } from 'react-router';

import { SUPPORT_LANGUAGES } from '../../../../constants';
import messages from '../../../app/i18n/fi.json';
import { ROUTES } from '../../../app/routes/constants';
import EditEventButtons from '../EditEventButtons';

it('matches snapshot', () => {
  const { container } = render(
    <MemoryRouter>
      <EditEventButtons
        dirty
        onClickLanguage={jest.fn()}
        selectedLanguage="fi"
      />
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

it('renders texts correctly', () => {
  const { rerender } = render(
    <MemoryRouter>
      <EditEventButtons
        dirty
        onClickLanguage={jest.fn()}
        selectedLanguage="fi"
      />
    </MemoryRouter>
  );

  expect(
    screen.queryByRole('button', {
      name: messages.editEvent.buttons.buttonBack,
    })
  ).toBeVisible();

  expect(
    screen.queryByText(messages.editEvent.buttons.textDirty)
  ).toBeVisible();

  rerender(
    <MemoryRouter>
      <EditEventButtons
        dirty={false}
        onClickLanguage={jest.fn()}
        selectedLanguage="fi"
      />
    </MemoryRouter>
  );

  expect(
    screen.queryByText(messages.editEvent.buttons.textDirty)
  ).not.toBeInTheDocument();

  Object.values(SUPPORT_LANGUAGES).forEach((lang) => {
    const langButton = screen.queryByRole('button', {
      name: messages.common.languages[lang],
    });
    expect(langButton).toBeVisible();
    expect(langButton).not.toBeDisabled();
  });
});

it('calls callback correctly when clicked', () => {
  const pushMock = jest.fn();
  jest.spyOn(Router, 'useHistory').mockReturnValue({ push: pushMock });
  const languageClickMock = jest.fn();
  render(
    <EditEventButtons
      dirty
      onClickLanguage={languageClickMock}
      selectedLanguage="fi"
    />
  );

  userEvent.click(
    screen.getByRole('button', {
      name: messages.editEvent.buttons.buttonBack,
    })
  );

  expect(pushMock).toHaveBeenCalledWith(ROUTES.HOME);

  Object.values(SUPPORT_LANGUAGES).forEach((lang, i) => {
    const langButton = screen.getByRole('button', {
      name: messages.common.languages[lang],
    });
    userEvent.click(langButton);
    expect(languageClickMock).toHaveBeenCalledTimes(i + 1);
    expect(languageClickMock).toHaveBeenCalledWith(lang);
  });
});
