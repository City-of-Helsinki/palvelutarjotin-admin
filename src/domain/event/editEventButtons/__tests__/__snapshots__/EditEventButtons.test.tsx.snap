// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="editEventButtons"
  >
    <div
      class="backButtonWrapper"
    >
      <button
        class="backButton"
      >
        <div
          class="backIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <rect
                height="24"
                width="24"
              />
              <polygon
                fill="currentColor"
                points="10 5.5 11.5 7 7.5 11 20 11 20 13 7.5 13 11.5 17 10 18.5 3.5 12"
              />
            </g>
          </svg>
        </div>
        <span>
          Tapahtumat
        </span>
      </button>
      <span
        class="dirtyText"
      >
        Sivulla on tallentamattomia muutoksia
      </span>
    </div>
    <div
      class="eventLanguageSelector"
    >
      <button
        class="languageButton isSelected"
        data-testid="eventLanguageSelector-suomi"
        type="button"
      >
        <div
          class="buttonIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          suomi
        </span>
      </button>
      <button
        class="languageButton"
        data-testid="eventLanguageSelector-ruotsi"
        type="button"
      >
        <div
          class="buttonIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          ruotsi
        </span>
      </button>
      <button
        class="languageButton"
        data-testid="eventLanguageSelector-englanti"
        type="button"
      >
        <div
          class="buttonIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm1 6v6.5h2V18H9v-1.5h2v-5H9V10h4zm-1.187-4a1.312 1.312 0 110 2.625 1.312 1.312 0 010-2.625z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <span>
          englanti
        </span>
      </button>
    </div>
  </div>
</div>
`;
